driver-core/src/main/java/com/datastax/driver/core/TypeCodec.java

```diff
@@ -930,6 +933,214 @@ public Date deserialize(ByteBuffer bytes) {
         }
     }
 
+    static class SimpleDateCodec extends TypeCodec<DateWithoutTime> {
+
+        public static final SimpleDateCodec instance = new SimpleDateCodec();
+        private static final Pattern IS_LONG_PATTERN = Pattern.compile("^-?\\d+$");
+
+        private static final String[] patterns = new String[] {
+                                                              "yyyy-MM-dd",
+                                                              "yyyy-MM-ddZ"
+        };
+
+        private SimpleDateCodec() {}
+
+        private static DateWithoutTime parseDate(String str, final String[] parsePatterns) {
+            SimpleDateFormat parser = new SimpleDateFormat();
+            parser.setLenient(false);
+            parser.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+            ParsePosition pos = new ParsePosition(0);
+            for (String parsePattern : parsePatterns) {
+                parser.applyPattern(parsePattern);
+                pos.setIndex(0);
+
+                Date date = parser.parse(str, pos);
+                if (date != null && pos.getIndex() == str.length()) {
+                    return DateWithoutTime.fromMillis(date.getTime());
+                }
+            }
+            throw new IllegalArgumentException("Unable to parse the date: " + str);
+        }
+
+        @Override
+        public DateWithoutTime parse(String value) {
+            if (IS_LONG_PATTERN.matcher(value).matches()) {
+                try {
+                    return DateWithoutTime.fromSimpleDate(Integer.parseInt(value));
+                } catch (NumberFormatException e) {
+                    throw new InvalidTypeException(String.format("Cannot parse date value from \"%s\"", value));
+                }
+            }
+
+            try {
+                return parseDate(value, patterns);
+            } catch (IllegalArgumentException e) {
+                throw new InvalidTypeException(String.format("Cannot parse date value from \"%s\"", value));
+            }
+        }
+
+        @Override
+        public String format(DateWithoutTime value) {
+            return value.toString();
+        }
+
+        @Override
+        public ByteBuffer serialize(DateWithoutTime value) {
+            return IntCodec.instance.serializeNoBoxing(value.getDays());
+        }
+
+        @Override
+        public DateWithoutTime deserialize(ByteBuffer bytes) {
+            return DateWithoutTime.fromSimpleDate(IntCodec.instance.deserializeNoBoxing(bytes));
```

Comment:
In the native protocol, DATE is an unsigned integer. `IntCodec` reads a signed integer, so we need to adjust the value:
```java
// The protocol encodes DATE as an _unsigned_ int with the epoch in the middle of the range (2^31).
// Because ByteBuffer does not have a `getUnsignedInt` method, we read those 4 bytes as a signed int
// and convert it. In addition, we shift the value to have the epoch at 0 like is usual in Java.
// These two methods handle the conversions.

private static int protocolToJava(int p) {
    return (p >= 0)
        ? Integer.MIN_VALUE + p
        : Integer.MAX_VALUE + p + 1;
}

private static int javaToProtocol(int j) {
    return (j >= 0)
        ? j - Integer.MIN_VALUE
        : j - Integer.MAX_VALUE - 1;
}
```
Use `protocolToJava` in `deserialize` and `javaToProtocol` in `serialize`.